### PR TITLE
Add new sensor that exposes the current files estimated print time

### PR DESCRIPTION
**Pull request by WolfeCub**
Already opened issue #97. Figured I'd try making the change myself. 

If I understand correctly this should be all that's needed. 

Let me know if I missed something.

Cheers!

### CODE REVIEW

**Comment by marcolivierarsenault** [comment]
Looking at the code, does that work? Have you tried that on your setup?

**Comment by WolfeCub** [comment]
image

Just gave it a test. Made a small tweak to use your `empty_result_when_not_printing` function and everything seems to be good. I tried changing the print and it correctly updated with the new estimated time as shown in the screenshot.

File: custom_components/moonraker/sensor.py [hunk 1]
```diff
@@ -178,6 +178,16 @@ class MoonrakerSensorDescription(SensorEntityDescription):
         icon="mdi:timer",
         device_class=SensorDeviceClass.TIMESTAMP,
     ),
+    MoonrakerSensorDescription(
```
Comment (marcolivierarsenault) [inline custom_components/moonraker/sensor.py:181]: can you add a test case for it somewhere in this function: https://github.com/marcolivierarsenault/moonraker-home-assistant/blob/258d2958de8d77bc65f9561d14b50c8926ead23f/tests/test_sensor.py#L80

File: custom_components/moonraker/sensor.py [hunk 2]
```diff
@@ -178,6 +178,16 @@ class MoonrakerSensorDescription(SensorEntityDescription):
...
+        key="file_print_estimate",
+        name="File Print Estimate",
```
Comment (marcolivierarsenault) [inline custom_components/moonraker/sensor.py:183]: Can we call this `Slicer Print Duration Estimate`

File: custom_components/moonraker/sensor.py [hunk 3]
```diff
@@ -178,6 +178,16 @@ class MoonrakerSensorDescription(SensorEntityDescription):
...
+        value_fn=lambda sensor: sensor.empty_result_when_not_printing(
+            sensor.coordinator.data["estimated_time"]
+        ),
+        subscriptions=[],
+        icon="mdi:timer",
+        device_class=SensorDeviceClass.DURATION,
```
Comment (marcolivierarsenault) [inline custom_components/moonraker/sensor.py:189]: We should add unit:
        unit=TIME_SECONDS,

**Comment by WolfeCub** [comment]
@marcolivierarsenault thanks for the feedback!

I've addressed all the comments. Let me know if there's anything else.

**Comment by marcolivierarsenault** [comment]
Do you only want the total estimate, or also the time left:
Right now only the top one shows up.
image

```python
MoonrakerSensorDescription(
        key="slicer_print_time_left_estimate",
        name="Slicer Print Time Left Estimate",
        value_fn=lambda sensor: sensor.empty_result_when_not_printing(
            sensor.coordinator.data["estimated_time"]
            - sensor.coordinator.data["status"]["print_stats"]["print_duration"]
        ),
        subscriptions=[("print_stats", "print_duration")],
        icon="mdi:timer",
        device_class=SensorDeviceClass.DURATION,
        unit=TIME_SECONDS,
    ),
```

For the second.

Should we add it?

**Comment by WolfeCub** [comment]
I was planning on just doing that calculation in a home assistant template since the duration is already there. Having a second sensor might be useful to others though, so it might be worth adding.

**Comment by marcolivierarsenault** [comment]
ok can you add it, then I will approve the PR.

Also, thanks a lot for your active contribution

**Review by marcolivierarsenault — COMMENTED** [review]
Sorry for the timing, 
But we just updated our doc infrastructure.

Can you go add documentation about your 2 sensors in https://raw.githubusercontent.com/marcolivierarsenault/moonraker-home-assistant/main/docs/entities/sensors.rst

**Comment by marcolivierarsenault** [comment]
You also have a failing CI to fix.

**Comment by WolfeCub** [comment]
@marcolivierarsenault no worries.

I added the doc and ran black on everything.

**Review by marcolivierarsenault — APPROVED** [review]
Looks good

**Comment by marcolivierarsenault** [comment]
Thanks a lot for your contribution

**Comment by WolfeCub** [comment]
Awesome. Thanks for the quick turnaround! 

Let me know when you cut a new release.

**Pull request merged**